src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -284,6 +284,26 @@ def train_from_config(cls, cfg: Union[DictConfig, dict]) -> Tuple[nn.Module, Tup
             dataloader_params=cfg.dataset_params.val_dataloader_params,
         )
 
+        test_loaders = {}
+        if "test_dataset_params" in cfg.dataset_params:
+            test_dataloaders = get_param(cfg, "test_dataloaders")
```

Comment:
I feel this is getting a bit all over the place...
test_dataloader_parameters are optional, they must fit the keys, test_dataset_params that are mappings...

I feel the form where we pass the dataset class name through the dataloader params might simplify all of this.

I.e, like in the pose recipes:
```
train_dataloader_params:
  dataset: COCOPoseEstimationDataset
  batch_size: 8
  num_workers: 8
  drop_last: True
  pin_memory: False
  shuffle: True
  collate_fn: YoloNASPoseCollateFN
```

I tend to think in general this is a preferred way, and think we should migrate at some point.